File: src/main/java/dark/leech/text/enities/BookEntity.java
package dark.leech.text.enities;

import java.io.Serializable;

import com.google.gson.annotations.SerializedName;

import lombok.Getter;
import lombok.Setter;

@Setter
@Getter
public class BookEntity implements Serializable {

    @SerializedName("book_id")
    private String id = "";

    @SerializedName("name")
    private String name;

    @SerializedName("author")
    private String author;

    @SerializedName("cover")
    private String cover;

    @SerializedName("url")
    private String url;

    @SerializedName("introduce")
    private String introduce;

    @SerializedName("web_source")
    private String webSource;

    @SerializedName("detail")
    private String detail;
}

File: engine/src/commonMain/kotlin/dev/haipham22/leechtext/entities/BookEntity.kt
package dev.haipham22.leechtext.entities

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/** Metadata sách (port từ enities/BookEntity.java). */
@Serializable
data class BookEntity(
    @SerialName("book_id") var id: String = "",
    @SerialName("name") var name: String? = null,
    @SerialName("author") var author: String? = null,
    @SerialName("cover") var cover: String? = null,
    @SerialName("url") var url: String? = null,
    @SerialName("introduce") var introduce: String? = null,
    @SerialName("web_source") var webSource: String? = null,
    @SerialName("detail") var detail: String? = null,
    /** True = Đang ra, false = Hoàn thành — field `ongoing` từ detail.js (vd truyenfull). */
    @SerialName("ongoing") var ongoing: Boolean? = null,
)
